core/src/main/java/org/apache/brooklyn/feed/CommandPollConfig.java

```diff
@@ -0,0 +1,201 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.feed;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.core.feed.PollConfig;
+import org.apache.brooklyn.feed.ssh.SshPollValue;
+import org.apache.brooklyn.util.collections.MutableList;
+import org.apache.brooklyn.util.collections.MutableMap;
+
+public class CommandPollConfig<T> extends PollConfig<SshPollValue, T, CommandPollConfig<T>> {
+
+    private Supplier<String> commandSupplier;
+    private List<Supplier<Map<String,String>>> dynamicEnvironmentSupplier = MutableList.of();
+
+    public static final Predicate<SshPollValue> DEFAULT_SUCCESS = new Predicate<SshPollValue>() {
+        @Override
+        public boolean apply(@Nullable SshPollValue input) {
+            return input != null && input.getExitStatus() == 0;
+        }};
+
+    public static <T> CommandPollConfig<T> forSensor(AttributeSensor<T> sensor) {
+        return new CommandPollConfig<T>(sensor);
+    }
+
+    public static CommandPollConfig<Void> forMultiple() {
+        return new CommandPollConfig<Void>(PollConfig.NO_SENSOR);
+    }
+
+    public CommandPollConfig(AttributeSensor<T> sensor) {
+        super(sensor);
+        super.checkSuccess(DEFAULT_SUCCESS);
+    }
+
+    public CommandPollConfig(CommandPollConfig<T> other) {
+        super(other);
+        commandSupplier = other.commandSupplier;
+    }
+    
+    /** @deprecated since 0.7.0; use {@link #getCommandSupplier()} and resolve just-in-time */
+    @Deprecated
+    public String getCommand() {
+        return getCommandSupplier().get();
+    }
+    public Supplier<String> getCommandSupplier() {
+        return commandSupplier;
+    }
+    
+    /** @deprecated since 0.7.0; use {@link #getEnvSupplier()} and resolve just-in-time */
+    @Deprecated
+    public Map<String, String> getEnv() {
+        return getEnvSupplier().get();
+    }
+    
+    @SuppressWarnings("unused")
+    public Supplier<Map<String,String>> getEnvSupplier() {
+        if (true) return new CombiningEnvSupplier(dynamicEnvironmentSupplier);
+        
+        // TODO Kept in case it's persisted; new code will not use this.
+        return new Supplier<Map<String,String>>() {
```

Comment:
Since you are renaming the class this no longer needs to be kept around. Could add a renaming rule for the anonymous class to `CombiningEnvSupplier`.